common-lib/src/main/scala/com/gu/mediaservice/lib/bbc/components/BBCImageProcessorsDependencies.scala

```diff
@@ -0,0 +1,43 @@
+package com.gu.mediaservice.lib.bbc.components
+
+import com.gu.mediaservice.lib.bbc.BBCImageProcessorConfig
+import play.api.Configuration
+
+import scala.collection.mutable
+import scala.concurrent.ExecutionContext
+
+object BBCImageProcessorsDependencies {
+  implicit val ec = ExecutionContext.global
+
+  private def memoizeOnce[A, B](f: A => B): A => B = new ((A) => B) {
```

Comment:
For me this memoisation logic could be confusing (were multiple configurations to be used for different processors, this will result in only the first one being initialised actually working). Having `save` be a Map[A, B] and only computing the output once for each different input might resolve this, assuming A correctly implements `Comparable`. 

That said this is logic that won't be used outside the BBC so I don't consider it as a blocker for merging.

Reply:
I agree, it makes sense if only it is used with the CommonConfiguration. The first version was a Map, but it has the potential to have some memory leak. For instance, if you let hot reload the config many times.
I will change the signature from ` ImageProcessorResources => MetadataStore` to `CommonConfig =>MetadataStore ` in order to be less confusing.